docs/interviewing_options.md

```diff
@@ -1 +1,134 @@
+- Can bias against people whom have family commitments (caregivers,
+  single parents, etc.)
+- Unintentionally discriminate based on age (favouring those whom might have
+  time to study, or have recently graduated from academia)
+- Creates a pressure cooker situation where candidates make mistakes[1]
+  - People whom might suffer from anxiety might perform well in these situations
+- Many of the questions asked in these formats are not real world type problems
+  and ultimately someone whom can invert a binary tree may not be able to
+  deliver on actual business problems faced by businesses
+- If candidate has potentially memorized the solution (from another interview
+  or seen it posted online) this again may not be indicative of their true
+  knowledge.
+- Unfair power imbalance between candidates and interviewers
+
+
+
+## Take-home tests (Homework, Challenges)
+
+Take-home tests are where a candidate is given an assignment to do on their own
+time and typically expected to return their completed assignment back in a set
+period time. this can be hours or days. Sometimes the results of the assignment
+become the topic of conversation during the on-site portion of the interview.
+
+#### Pros
+
+- Allows the candidate to work on their own time, space, environment setup, etc.
+- Candidate can use Google, and other online (legit) resources like they might
+  at a typical job
+- Can be done during non work hours allowing candidates more freedom
+- Less anxiety involved for candidate as no one is watching them
+- Certain types of tests might be conducive to automated testing to check
+  results and validity
+- Again this can be a cheap way to screen candidates as they don't need to come
+  on-site, and act as a stopper for those who are not super serious about
+  actually interviewing
+- Removes the power imbalance that can occur in whiteboard style interview.
+
+#### Cons
+
+- Possibility of candidates plagiarizing work
+- Can bias against people whom have family commitments (caregivers,
+  single parents, etc.) depending on the level of effort needed
+- Again can unintentionally discriminate based on age (favouring those whom
+  might have more free time and can iterate on better solutions)
+- Depending on the complexity of the assignment it may require time for someone
+  to grade and properly evaluate. Depending on the volume of candidates this
+  can be a huge problem.
+
+
+## Bring your own personal code or project
+
+Bringing your own personal code or projects is another method similar to a take
+type of test. The main difference here is that this is something the candidate
+has built vs being told to do. This could be a open source project which they
+have contributed to or maybe a smaller hobby type project. During the interview
+process
+
+#### Pros
+
+- Allows the candidate to work on their own time, space, environment setup, etc.
+- Can be done during non work hours allowing candidates more freedom
```

Comment:
I would say that the work is already done, so there is no extra time commitment on the part of the candidate other than maybe to review the code before they discuss it if it has been awhile since they looked at it.

Reply:
👍